Sources/PayoneerCheckout/Interface/Checkout.swift

```diff
@@ -0,0 +1,117 @@
+@objc public class Checkout: NSObject {
+    public let configuration: CheckoutConfiguration
+    private(set) weak var presenter: UIViewController?
+    private(set) var paymentListViewController: UIViewController?
+    private(set) var paymentCompletionBlock: ((_ result: CheckoutResult) -> Void)?
+    private let chargePresetService: ChargePresetServiceProtocol
+
+    /// Initializes a `Checkout` with the given configuration.
+    /// - Parameters:
+    ///   - configuration: The configuration object to be used.
+    @objc public init(configuration: CheckoutConfiguration) {
+        self.configuration = configuration
+        self.chargePresetService = ChargePresetService(riskProviders: configuration.riskProviders)
+        CheckoutAppearance.shared = configuration.appearance
+    }
+
+    init(configuration: CheckoutConfiguration, chargePresetService: ChargePresetServiceProtocol) {
+        self.configuration = configuration
+        self.chargePresetService = chargePresetService
+        CheckoutAppearance.shared = configuration.appearance
+    }
+}
+
+// MARK: - Operations
+
+@objc public extension Checkout {
+    /// Presents the checkout UI.
+    /// - Parameters:
+    ///   - presenter: The view controller that will present the checkout UI.
+    ///   - completion: The block to execute after the operation is complete and the UI is dismissed.
+    ///
+    ///     This completion block takes the following parameter:
+    ///   - result: An object containing relevant information about the result of the operation.
+    func presentPaymentList(from presenter: UIViewController, completion: @escaping (_ result: CheckoutResult) -> Void) {
+        let paymentListViewController = PaymentListViewController(listResultURL: configuration.listURL, riskProviders: configuration.riskProviders, delegate: self)
+
+        self.presenter = presenter
+        self.paymentCompletionBlock = completion
+        self.paymentListViewController = paymentListViewController
+
+        let navigationController = UINavigationController(rootViewController: paymentListViewController)
+
+        if let customAccentColor = configuration.appearance.accentColor {
+            navigationController.view.tintColor = customAccentColor
+        }
+
+        presenter.present(navigationController, animated: true, completion: nil)
+    }
+
+    /// Charges a preset account.
+    /// - Parameters:
+    ///   - completion: The block to execute after the operation is complete. It includes an object with relevant information about the result of the operation.
+    ///
+    ///     This completion block takes the following parameter:
+    ///   - result: An object containing relevant information about the result of the operation.
+    func chargePresetAccount(from presenter: UIViewController, completion: @escaping (_ result: CheckoutResult) -> Void) {
```

Comment:
1. In this method `from` is not descriptive and I couldn't find any good naming for external parameter name.
2. Could you add a documentation for `presenter` parameter, in that special case it is not very clear for a merchant why should he use `presenter` here.

```suggestion
    func chargePresetAccount(presenter: UIViewController, completion: @escaping (_ result: CheckoutResult) -> Void) {
```

Reply:
Good catch, I missed this when I added this argument.

Reply:
https://github.com/optile/checkout-ios/pull/185/commits/54fc00e0ea301d647b946e0d812974a962541b26